packages/at_client/lib/src/service/sync_service_impl.dart

```diff
@@ -429,23 +438,38 @@ class SyncServiceImpl implements SyncService, AtSignChangeListener {
   }
 
   /// Syncs the cloud secondary changes to local secondary.
-  Future<List<KeyInfo>> _syncFromServer(
-      int serverCommitId,
-      int lastReceivedServerCommitId,
-      List<CommitEntry> uncommittedEntries) async {
+  Future<List<KeyInfo>> _syncFromServer(int serverCommitId,
+      int lastReceivedServerCommitId, List<CommitEntry> uncommittedEntries,
+      {int? localCommitIdBeforeSync}) async {
     // Iterates until serverCommitId is greater than lastReceivedServerCommitId.
     // replacing localCommitId with lastReceivedServerCommitId fixes infinite loop issue
     // in certain scenarios e.g server has a commit entry that need not be synced on client side,
     // server has delete commit entry and the key is not present on local keystore
     List<KeyInfo> keyInfoList = [];
     try {
+      int? skipDeletesUntil;
```

Comment:
Should we move this to a private method  getAndPopulateSkipDeletesUntil() so that we can write tests too